lib/awful/ewmh.lua

```diff
@@ -291,7 +295,8 @@ function ewmh.geometry(c, context, hints)
             props.honor_padding = beautiful.maximized_honor_padding ~= false
         end
 
-        if original_context == "fullscreen" and beautiful.fullscreen_hide_border ~= false then
+        if (original_context == "fullscreen" and beautiful.fullscreen_hide_border ~= false) or
+           (original_context == "maximized" and beautiful.maximized_hide_border == true) then
```

Comment:
Why `~= false` for one and `== true` for the other?

Reply:
The fullscreen one is enabled by default, so if it's not set or set to true, it's on. With maximized it's only on when you set it to true explicitly in your config.

Reply:
@alfunx Specifically, the `false` is there so that `nil` is not treated as `false` (`nil ~= false` is true)